### PR TITLE
'View All' Bugfix on Notifications page

### PR DESCRIPTION
**Pull request by jeanetteobr**
### Verification
This PR addresses a bug in prod on the Notifications page that causes an empty query string to be sent to the Notifications endpoint on the selection of the 'View All' filter. 

**Before:**
DMG1fsE3SY

**After:**
GFE0niCKt2

Jira Issue: https://issues.redhat.com/browse/AA-238

### CODE REVIEW

File: src/Utilities/useQueryParams.js [hunk 1]
```diff
@@ -28,6 +29,12 @@ export const useQueryParams = initial => {
             /* v1 api reducers */
             case 'SET_OFFSET':
             case 'SET_SEVERITY':
+                if (value.severity === '') {
```
Comment (brumik) [inline src/Utilities/useQueryParams.js:32]: You cannot do this, this way you break the `SET_OFFSET` rule here.

File: src/Utilities/useQueryParams.js [hunk 2]
```diff
@@ -28,6 +29,12 @@ export const useQueryParams = initial => {
...
+                    const { severity: ignored, ...rest } = state;
```
Comment (brumik) [inline src/Utilities/useQueryParams.js:33]: We don't need `ignored` anymore, there was a rule added in eslint for destructuring + unused

File: src/Utilities/useQueryParams.js [hunk 3]
```diff
@@ -94,7 +101,9 @@ export const useQueryParams = initial => {
         dispatch,
         setLimit: limit => dispatch({ type: 'SET_LIMIT', value: { limit }}),
         setOffset: offset => dispatch({ type: 'SET_OFFSET', value: { offset }}),
-        setSeverity: severity => dispatch({ type: 'SET_SEVERITY', value: { severity }}),
+        setSeverity: (severity) => {
```
Comment (brumik) [inline src/Utilities/useQueryParams.js:104]: I think here no functionality changed except not returning the dispatch return value?

File: src/Utilities/useQueryParams.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (brumik) [inline src/Utilities/useQueryParams.js:1]: Forgot eslint rule

**Review by brumik — CHANGES_REQUESTED** [review]

**Comment by brumik** [comment]
@kialam I remember when we could break this: when we were converting maybe from `urlMappedQueryParams` to `queryParams` only. Since in the `urlMappedQuerryParams` I was removing the empty params :thinking:

File: src/Utilities/useQueryParams.js (hunk 1 above)
Comment (jeanetteobr) [reply src/Utilities/useQueryParams.js:32]: Could you please elaborate on how the logic on line 32 would break the `SET_OFFSET` rule?

File: src/Utilities/useQueryParams.js (hunk 2 above)
Comment (jeanetteobr) [reply src/Utilities/useQueryParams.js:33]: We are doing something similar on line 21 with `SET_LIMIT`...

File: src/Utilities/useQueryParams.js (hunk 3 above)
Comment (jeanetteobr) [reply src/Utilities/useQueryParams.js:104]: I was console logging some values here and neglected to revert the function to its original state...

File: src/Utilities/useQueryParams.js (hunk 2 above)
Comment (brumik) [reply src/Utilities/useQueryParams.js:32]: Yes, it was needed before, but we changed the eslint rules a week or two back, so now we don't need it

File: src/Utilities/useQueryParams.js (hunk 1 above)
Comment (brumik) [reply src/Utilities/useQueryParams.js:31]: Hm, hard to explain for me so sorry if it baroquesq:

When you write a list of cases then it will jump off the only on break or return from the switch block. So in this example right now from `SET_OFFSET` to `SET_ROOT_WORKFLOWS_AND_JOBS` were using the same line of code: `return { ...state, ...value };` since there were no other `return` or `break` statements.

Now you modified it sonow the `SET_OFFSET` which is before `SET_SEVERITY` uses the logic what you wrote for `SET_SEVERITY`. 

So basically if there is no break or return after a case the next case code is also run. 

So for example you can (and maybe should) delete your `return { ...state, ...value };` line and just leave the `if` block, since you won't have any return so the last `return { ...state, ...value };` would run.

Comment (brumik) [reply src/Utilities/useQueryParams.js:31]: So, yeah, I didn't tested the code and it may work, but it is not safe to leave it like this I think. My solution would be to move the `SET_SEVERITY` case block under `SET_ROOT_WORKLOFS_AND_JOBS` code block.

Comment (brumik) [reply src/Utilities/useQueryParams.js:31]: And sorry that I didn't explain it before, it was a mistake I made in a hurry.

File: src/Utilities/useQueryParams.js (hunk 2 above)
Comment (brumik) [reply src/Utilities/useQueryParams.js:32]: It's not actually is a problem, I just don't like extra character typing :D but you can leave it

**Review by brumik — APPROVED** [review]
I approve, the bug is gone!

**Comment by brumik** [comment]
I guess we need to test the `queryParam` hook better in jest, if you want to try you can look for my `useApi.test.js` where I test a hook. It's pretty simple :) But that is outside of the scope of PR so this is good to go
